pkg/controller/statusmanager/status_manager.go

```diff
@@ -49,11 +57,26 @@ type StatusManager struct {
 
 	failing [maxStatusLevel]*configv1.ClusterOperatorStatusCondition
 
-	daemonSets     []types.NamespacedName
-	deployments    []types.NamespacedName
+	daemonSets     map[types.NamespacedName]*daemonsetState
+	deployments    map[types.NamespacedName]*deploymentState
 	relatedObjects []configv1.ObjectReference
 }
 
+// daemonsetState is the internal state we use to check if a rollout has
+// stalled. Since we don't get a last-changed time from the apiserver, we
+// need to synthesize it ourselves. This means that we may flap between
+// Degraded if the operator restarts.
```

Comment:
Clayton hates it when operator statuses flap on startup... we could put the last change time into the Progressing message and then parse it back out on each update...

Reply:
Argh. Fine. We need to know the last change time of each daemonset/deployment individually. I suppose we could marshall it as yaml so it's reasonably readable.

Reply:
You can just add it to the individual lines (`"Deployment openshift-sdn/sdn is not available (awaiting 1 nodes since 2019-10-22T14:22:00)"`.
Or use an annotation if you don't like putting it in the status.